### PR TITLE
Adding a test with the driver of SICKTim561Eth via rawlog-grabber

### PR DESCRIPTION
**Pull request by alibenD**
## Changed apps/libraries

* modified-libX
* modified-appY
* ...

## PR Description

*  This PR Closes issue #628, registered `SICKTim561Eth` in `registerAllClasses.cpp` and created new `ini` for `SICKTim561Eth`. Developers can test a physical device without building sample problem, just via rawlog-grabber.

---

I acknowledge to have:
* Read the [`CONTRIBUTING`](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md) page
* Updated [`doc/doxygen-pages/changeLog_doc.h`](https://github.com/MRPT/mrpt/blob/master/doc/doxygen-pages/changeLog_doc.h) to describe these changes (if applicable)
* Updated [`AUTHORS`](https://github.com/MRPT/mrpt/blob/master/AUTHORS) (if applicable)

(Notify: @MRPT/owners )

### CODE REVIEW

**Comment by alibenD** [comment]
- Tested it on a physical device(Sick Tim561)
- Wrote a setting file ''SICK_TIM_56x.ini" for rawlog_grabber
- Tested it via rawlog_grabber.

The device works well.

File: share/mrpt/config_files/rawlog-grabber/SICK_TIM_56x.ini [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+pose_x			= 0		// Laser range scaner 3D position in the robot (meters)
+pose_y			= 0
+pose_z			= 0.31
+pose_yaw		= 0		// Angles in degrees
+pose_pitch		= 0
+pose_roll		= 0
+
+m_ip = 192.168.0.1  // Default IP address
+m_port = 2111       // Default IP Port
+m_process_rate = 15 // Default Herzt
```
Comment (jlblancoc) [inline share/mrpt/config_files/rawlog-grabber/SICK_TIM_56x.ini]: Thanks for the PR! :+1: 

Just a comment on these `m_*` variable names: for convention and homogeneity with the rest of driver classes, variables in configuration files don't use to have that prefix. Please, take a look at other classes, and update the PR if you see it easy to do....

**Comment by alibenD** [comment]
Thank you for comment. It should be updated in today.

File: share/mrpt/config_files/rawlog-grabber/SICK_TIM_56x.ini (hunk 1 above)
Comment (alibenD) [reply share/mrpt/config_files/rawlog-grabber/SICK_TIM_56x.ini]: Thanks for the comment. The `ini` file has been updated without prefix, and the PR has been updated. What else can I do for this PR? Please feel free to leave a message at here, I am so glad to do it. Thanks!

**Comment by jlblancoc** [comment]
Ok! I now see that the variable names in the INI file match those expected given the names here: 
https://github.com/MRPT/mrpt/blob/master/libs/hwdrivers/src/CSICKTim561Eth_2050101.cpp#L83-L85

One more final question: "process_rate" is set to 15... if that is exactly the rate at which the LiDAR acquires scans, I would recommend you testing with a **larger** rate, such that, sometimes, the "doProcess()" method won't collect a scan, others will do. In this way, we would be safer of not "falling behind" in case of any delay in data processing. I hope I explained it well... 

Please, give it a try with a larger process_rate in the example INI file, for example, >=20, and let me know if it works before merging.

**Comment by alibenD** [comment]
Hi Professor Blanco, I might misunderstand `process rate` and `scanning rate`. 

And I have tested a process rate = 25,50,100, respectively, it looks working. The info from a terminal is as follows:
```
rawlog-grabber - Part of the MRPT
 MRPT C++ Library: MRPT 1.9.9 - Sources timestamp: 2017-12-08 04:14:21 UTC
-------------------------------------------------------------------

Output rawlog filename: ./dataset_2017-12-11_11h02m41s.rawlog
External image storage: ./dataset_2017-12-11_11h02m41s_Images

[thread_CSICKTim561Tim] Starting... at 100 Hz

Press any key to exit program
[C2DRangeFinderAbstract|DEBUG|11:02:42.8287] read : 8
[C2DRangeFinderAbstract|DEBUG|11:02:42.8290] message : sFA 11
[C2DRangeFinderAbstract|DEBUG|11:02:43.8322] read : 29
[C2DRangeFinderAbstract|DEBUG|11:02:43.8323] message : sRA SerialNumber 8 17230872
[C2DRangeFinderAbstract|DEBUG|11:02:44.8353] read : 29
[C2DRangeFinderAbstract|DEBUG|11:02:44.8354] message : sRA FirmwareVersion 5 V2.54
[C2DRangeFinderAbstract|DEBUG|11:02:45.8375] read : 21
[C2DRangeFinderAbstract|DEBUG|11:02:45.8377] message : sRA SCdevicestate 1
[C2DRangeFinderAbstract|DEBUG|11:02:45.8862] read : 100
``` 

About the scanning rate, just for your reference, in the official manual of Tim561 downloaded from [Tim561 Mannual](https://www.sick.com/sg/en/detection-and-ranging-solutions/2d-lidar-sensors/tim5xx/tim561-2050101/p/p369446), `scanning rate` is a constant parameter, no more option rate can be chosen.

**Comment by jlblancoc** [comment]
Thanks! 

> About the scanning rate, just for your reference, in the official manual of Tim561 downloaded from Tim561 Mannual, scanning rate is a constant parameter, no more option rate can be chosen.

Yes, most scanners have a fixed scan rate. But what I meant was that if the software driver is able to process one incoming scan at a time, it should run at the scan rate as the bare *minimum* rate... anything above will provide some "safety margin". 

All good, merging! 👍

**Pull request merged**